### ISSUE TITLE
`dask.array.nanmean()` generates unstable name (hash)

### ISSUE THREAD

**Issue by estebanag** [dask/dask#2988]
#### Code Sample

```python
import dask.array as da
import numpy as np

x = da.ones((5, 5), chunks=(2, 2))
x = da.nanmean(x, axis=0)
print(x.name)
```

#### Problem Description
Running the above sample three times outputs:
```
mean_agg-aggregate-8157871966f12f849b40951c495fef6f
mean_agg-aggregate-c6939e3b5eb3290f17252b895e10c7c3
mean_agg-aggregate-17e2ac663f1cdddf5b6fc02b1ad0b889
```
which shows the dask name (hash) is unstable.

#### Expected Output
For processing large datasets, it's convenient to have a stable hash name in order to save intermediate results or compare them among developers. 
> Workaround: use `dask.array.mean()` instead of `dask.array.nanmean()`

**Comment by jakirkham** [comment]
Please included how you created your test environment (e.g. `conda env export` if using `conda`).

**Comment by estebanag** [comment]
In order to get those results, I created a fresh virtual environment based on [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/) and installed dask using pip, as follows:
```bash
$ mkvirtualenv dask_test
$ pip install dask[complete]
```

**Comment by estebanag** [comment]
Here (Python 3.5.2):
```
click==6.7
cloudpickle==0.5.2
dask==0.16.0
distributed==1.20.2
HeapDict==1.0.0
locket==0.2.0
msgpack-python==0.4.8
numpy==1.13.3
pandas==0.21.0
partd==0.3.8
psutil==5.4.2
python-dateutil==2.6.1
pytz==2017.3
six==1.11.0
sortedcontainers==1.5.9
tblib==1.3.2
toolz==0.8.2
tornado==4.5.2
zict==0.1.3
```

**Comment by jcrist** [comment]
Thanks for the excellent bug report, this should be fixed in #3024.

```python
import dask
import dask.array as da

x = da.ones((5, 5), chunks=(5, 5))

x2 = da.nanmean(x, axis=0)
print(x2.name)
(dask) jcrist dask $ python test.py
mean_agg-aggregate-307d3de24a463578cdbedbb88a1b7461
(dask) jcrist dask $ python test.py
mean_agg-aggregate-307d3de24a463578cdbedbb88a1b7461
```
